### PR TITLE
feat: 404

### PR DESCRIPTION
**Pull request by PeeyushPrashant**
- user gets directed to 404 page whenever they try accessing routes which are not defined.
- added explore btn in Navbar
- refactored search functionality

Screenshot (131)

### CODE REVIEW

File: src/pages/Error/Error.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    const navigate=useNavigate();
+  return(
+     <>
+     <NavBar/>
+     <div className="main">
+     <div class="error-block flex-col">
+         <section className="error-status flex-row">
+            <div class="error-code">4</div>
+            <i class="far fa-question-circle fa-spin rotate-icon"></i>
+            <div class="error-code">4</div>
```
Comment (anuj-kumary) [inline src/pages/Error/Error.jsx]: Change class to className

**Review by anuj-kumary — APPROVED** [review]

**Review by anuj-kumary — COMMENTED** [review]
Suggestion:-
When you are on 404-page hide the cart and wishlist icon or you can hide the badge

File: src/App.js [hunk 2]
```diff
@@ -43,6 +44,7 @@ function App() {
         <Route path="/checkout" element={<CheckOut />} />
         <Route path="/order" element={<Orders />} />
```
Comment (Rohanmond) [inline src/App.js:45]: could add as a private route

**Review by Rohanmond — APPROVED** [review]
folder structure looks good and code looks good to merge

File: src/App.js (hunk 2 above)
Comment (PeeyushPrashant) [reply src/App.js:45]: will do it,thanks

Comment (PeeyushPrashant) [reply src/App.js:45]: user can go to checkout & Orders page only through Cart page, which is already a private route

**Pull request merged**